### PR TITLE
Add directive support for fab-dropdown

### PR DESCRIPTION
**Pull request by reshaw**
-Add directives to support i18n for fab-dropdown
-Add demo of new directive-supported fab-dropdown to app.component
-Minor cleanup of directives for fab-combo-box

### CODE REVIEW

File: libs/fabric/src/lib/components/combo-box/base-combo-box.component.ts [hunk 1]
```diff
@@ -20,7 +20,6 @@ import { ComboBoxOptionsDirective } from './directives/combo-box-options.directi
 
 export abstract class FabBaseComboBoxComponent extends ReactWrapperComponent<IComboBoxProps>
   implements OnInit, AfterContentInit {
-  @ContentChild(ComboBoxOptionDirective) readonly optionsDirective?: ComboBoxOptionDirective;
```
Comment (bengry) [inline libs/fabric/src/lib/components/combo-box/base-combo-box.component.ts]: Why was this removed?

File: libs/fabric/src/lib/components/dropdown/directives/dropdown-options.directive.ts [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+import { DropdownOptionDirective } from './dropdown-option.directive';
+import { IDropdownOption } from 'office-ui-fabric-react';
+
+/**
+ * Wrapper directive for creating multiple DropdownOptions
+ */
+@Directive({ selector: 'fab-dropdown > options' })
+export class DropdownOptionsDirective {
+  @ContentChildren(DropdownOptionDirective) readonly directiveItems: QueryList<DropdownOptionDirective>;
```
Comment (bengry) [inline libs/fabric/src/lib/components/dropdown/directives/dropdown-options.directive.ts]: maybe make this optional (and verify it exists in the `items` getter below, to make sure we catch the empty case:
```html
<fab-dropdown>
  <options></options>
</fab-dropdown>
```

File: libs/fabric/src/lib/components/dropdown/directives/dropdown-options.directive.ts [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+
+  get items() {
+    return this.directiveItems.map<IDropdownOption>(directiveItem => {
+      return {
+        key: directiveItem.optionKey,
+        text: directiveItem.text,
```
Comment (bengry) [inline libs/fabric/src/lib/components/dropdown/directives/dropdown-options.directive.ts]: WDYT about replacing these with spreading (`...`) the `directiveItem`? Except for the `key` -> `optionKey` of course.

File: libs/fabric/src/lib/components/combo-box/base-combo-box.component.ts (hunk 1 above)
Comment (reshaw) [reply libs/fabric/src/lib/components/combo-box/base-combo-box.component.ts]: I'm the one who put it there in the first place. It doesn't do anything.

File: libs/fabric/src/lib/components/dropdown/directives/dropdown-options.directive.ts (hunk 3 above)
Comment (reshaw) [reply libs/fabric/src/lib/components/dropdown/directives/dropdown-options.directive.ts]: Style preference on my end. I don't like using the … unless it's exhaustive for all items.

File: libs/fabric/src/lib/components/combo-box/base-combo-box.component.ts (hunk 1 above)
Comment (bengry) [reply libs/fabric/src/lib/components/combo-box/base-combo-box.component.ts]: Ok then 👍.

See the rest of my comments, but looks good. Feel free to merge when you're happy with it. The most important thing is to not break existing users that don't use the declarative API (but instead the imperative `options` one). Did you check that these still work?

File: libs/fabric/src/lib/components/dropdown/directives/dropdown-options.directive.ts (hunk 3 above)
Comment (bengry) [reply libs/fabric/src/lib/components/dropdown/directives/dropdown-options.directive.ts]: You could do the following:
```typescript
return this.directiveItems.map<IDropdownOption>(({ optionKey, ...otherDirectiveProps }) => ({
  key: optionKey,
  ...otherDirectiveProps
}))
```

With simple object destructuring you can get omitting props pretty easily, and then you have all props in `otherDirectiveProps`, except for the ones that you explicitly need to rename.

WDYT?

File: libs/fabric/src/lib/components/dropdown/dropdown.component.ts [hunk 4]
```diff
@@ -61,8 +64,9 @@ import { ISelectableDroppableTextProps, ISelectableOption } from 'office-ui-fabr
   styles: ['react-renderer'],
   changeDetection: ChangeDetectionStrategy.OnPush,
 })
-export class FabDropdownComponent extends ReactWrapperComponent<IDropdownProps> implements OnInit {
+export class FabDropdownComponent extends ReactWrapperComponent<IDropdownProps> implements OnInit, AfterContentInit {
   @ViewChild('reactNode') protected reactNodeRef: ElementRef;
+  @ContentChild(DropdownOptionsDirective) readonly dropdownOptionsDirective?: DropdownOptionsDirective;
```
Comment (bengry) [inline libs/fabric/src/lib/components/dropdown/dropdown.component.ts]: Please make sure that the imperative syntax of `options` still works as it did before, in addition to the declarative syntax.

Also, what happens when both syntaxes are used at once?

Comment (reshaw) [reply libs/fabric/src/lib/components/dropdown/dropdown.component.ts]: Interesting thought. I actually did test this with combo-box, but not both at the same time, just one at a time. I'll test and get back to you.

But yes, the old style works fine.

File: libs/fabric/src/lib/components/dropdown/directives/dropdown-options.directive.ts (hunk 3 above)
Comment (reshaw) [reply libs/fabric/src/lib/components/dropdown/directives/dropdown-options.directive.ts]: This seems fine to my sensibilities, and I agree with your logic. I'll make the change.

File: libs/fabric/src/lib/components/combo-box/base-combo-box.component.ts (hunk 1 above)
Comment (reshaw) [reply libs/fabric/src/lib/components/combo-box/base-combo-box.component.ts]: Yes! Imperative operation still works as before. When both are used simultaneously, though, the declarative one 'wins'. I'll add comments to the directives to indicate this.

File: libs/fabric/src/lib/components/dropdown/directives/dropdown-options.directive.ts (hunk 2 above)
Comment (reshaw) [reply libs/fabric/src/lib/components/dropdown/directives/dropdown-options.directive.ts]: The map handles this correctly. You just wind up with a slector that has no dropdown options.

**Comment by bengry** [comment]
@reshaw Note that you can also directly make a PR from within this repo. Since you're in the Commercial Stores team you have the permissions to create branches etc.

Will merge this once CLA reports the status, no idea why it didn't yet.

File: libs/fabric/src/lib/components/dropdown/dropdown.component.ts (hunk 4 above)
Comment (reshaw) [reply libs/fabric/src/lib/components/dropdown/dropdown.component.ts]: Tested. Both work in isolation (just-declarative, or just-imperative). If both are used at the same time, the declarative version 'wins'.
